internal/tescale/scale.go

```diff
@@ -0,0 +1,268 @@
+	return &IDPool{
+		nhIndex:  base,
+		nhgIndex: base,
+	}
+}
+
+// NextNHID returns the next NHID
+func (p *IDPool) NextNHID() uint64 {
+	p.rw.Lock()
+	defer p.rw.Unlock()
+
+	p.nhIndex++
+	return p.nhIndex
+}
+
+// NextNHGID returns the next NHGID
+func (p *IDPool) NextNHGID() uint64 {
+	p.rw.Lock()
+	defer p.rw.Unlock()
+
+	p.nhgIndex++
+	return p.nhgIndex
+}
+
+// VRFConfig holds NH, NHG and IPv4 entries for the VRF.
+type VRFConfig struct {
+	Name      string
+	NHs       []fluent.GRIBIEntry
+	NHGs      []fluent.GRIBIEntry
+	V4Entries []fluent.GRIBIEntry
+}
+
+// Param TE holds scale parameters.
+type Param struct {
+	V4TunnelCount         int
+	V4TunnelNHGCount      int
+	V4TunnelNHGSplitCount int
+	EgressNHGSplitCount   int
+	V4ReEncapNHGCount     int
+}
+
+// BuildVRFConfig creates scale new scale VRF configurations.
+func BuildVRFConfig(dut *ondatra.DUTDevice, egressIPs []string, param Param) []*VRFConfig {
+	v4TunnelIPAddrs := NewIPPool(iputil.GenerateIPs(V4TunnelIPBlock, param.V4TunnelCount))
+	v4VIPAddrs := NewIPPool(iputil.GenerateIPs(V4VIPIPBlock, (param.V4TunnelNHGCount*param.V4TunnelNHGSplitCount)+2))
+	v4EgressIPAddrs := NewIPPool(egressIPs)
+
+	defaultVRF := deviations.DefaultNetworkInstance(dut)
+	vrfTConf := &VRFConfig{Name: VRFT}
+	vrfRConf := &VRFConfig{Name: VRFR}
+	vrfRDConf := &VRFConfig{Name: VRFRD}
+	vrfDefault := &VRFConfig{Name: defaultVRF}
+	idPool := NewIDPool(10000)
+
+	// VRF_T:
+
+	nhgID := idPool.NextNHGID()
+	nhID := idPool.NextNHID()
+	nhgRedirectToVrfR := nhgID
+	// build backup NHG and NH.
+	vrfDefault.NHs = append(vrfDefault.NHs,
+		fluent.NextHopEntry().WithIndex(nhID).WithNetworkInstance(defaultVRF).WithNextHopNetworkInstance(VRFR),
+	)
+	vrfDefault.NHGs = append(vrfDefault.NHGs,
+		fluent.NextHopGroupEntry().WithID(nhgRedirectToVrfR).AddNextHop(nhID, 1).WithNetworkInstance(defaultVRF),
+	)
+
+	// Build IPv4 entry and related NHGs and NHs.
+	// * Mapping tunnel IP per the IP -> NHG ratio
+	// * Each NHG has unique NHs.
+	// * Each NHG has the same backup to Repair VRF.
+	tunnelNHGRatio := param.V4TunnelCount / param.V4TunnelNHGCount
+	for idx, ip := range v4TunnelIPAddrs.AllIPs() {
+		nhgEntry := fluent.NextHopGroupEntry().WithID(nhgID).WithNetworkInstance(defaultVRF).WithBackupNHG(nhgRedirectToVrfR)
+		if idx%tunnelNHGRatio == 0 {
+			nhgID = idPool.NextNHGID()
+
+			// Build NHs and link NHs to NHG.
+			for i := 0; i < param.V4TunnelNHGSplitCount; i++ {
+				vip := v4VIPAddrs.NextIP()
+				nhID = idPool.NextNHID()
+				vrfDefault.NHs = append(vrfDefault.NHs,
+					fluent.NextHopEntry().WithIndex(nhID).WithNetworkInstance(defaultVRF).WithIPAddress(vip),
+				)
+				nhgEntry = nhgEntry.AddNextHop(nhID, 1)
+			}
+			vrfDefault.NHGs = append(vrfDefault.NHGs, nhgEntry)
+		}
+
+		// Build IPv4 entry
+		vrfTConf.V4Entries = append(vrfTConf.V4Entries,
+			fluent.IPv4Entry().WithPrefix(ip).WithNextHopGroup(nhgID).WithNetworkInstance(VRFT).WithNextHopGroupNetworkInstance(defaultVRF),
```

Comment:
it should be fluent.IPv4Entry().WithPrefix(ip + "/32" ).

The client is publishing prefixes with no prefix length portion set and that is causing the failures.